pkg/resources/user_password_policy_attachment.go

```diff
@@ -0,0 +1,169 @@
+	"github.com/Snowflake-Labs/terraform-provider-snowflake/pkg/sdk"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+var userPasswordPolicyAttachmentSchema = map[string]*schema.Schema{
+	"user_name": {
+		Type:        schema.TypeString,
+		Required:    true,
+		ForceNew:    true,
+		Description: "User name of the user you want to attach the password policy to",
+	},
+	"password_policy_database": {
+		Type:        schema.TypeString,
+		Required:    true,
+		ForceNew:    true,
+		Description: "Database name where the password policy is stored",
+	},
+	"password_policy_schema": {
+		Type:        schema.TypeString,
+		Required:    true,
+		ForceNew:    true,
+		Description: "Schema name where the password policy is stored",
+	},
+	"password_policy_name": {
+		Type:        schema.TypeString,
+		Required:    true,
+		ForceNew:    true,
+		Description: "Non-qualified name of the password policy",
+	},
+}
+
+func UserPasswordPolicyAttachment() *schema.Resource {
+	return &schema.Resource{
+		Description: "Specifies the password policy to use for the current account. To set the password policy of a different account, use a provider alias.",
+
+		Create: CreateUserPasswordPolicyAttachment,
+		Read:   ReadUserPasswordPolicyAttachment,
+		Delete: DeleteUserPasswordPolicyAttachment,
+
+		Schema: userPasswordPolicyAttachmentSchema,
+
+		Importer: &schema.ResourceImporter{
+			StateContext: func(ctx context.Context, d *schema.ResourceData, m interface{}) ([]*schema.ResourceData, error) {
+				parts := strings.Split(d.Id(), helpers.IDDelimiter)
+				if len(parts) != 4 {
+					return nil, fmt.Errorf("id should be in the format 'database|schema|password_policy|user_name|roles', but I got '%s'", d.Id())
+				}
+				passwordPolicyDatabase := sdk.NewAccountIdentifierFromFullyQualifiedName(parts[0])
+				passwordPolicySchema := sdk.NewAccountIdentifierFromFullyQualifiedName(parts[1])
+				passwordPolicyName := sdk.NewAccountIdentifierFromFullyQualifiedName(parts[2])
+				userName := sdk.NewAccountObjectIdentifierFromFullyQualifiedName(parts[3])
+				if err := d.Set("password_policy_database", passwordPolicyDatabase.Name()); err != nil {
+					return nil, err
+				}
+				if err := d.Set("password_policy_schema", passwordPolicySchema.Name()); err != nil {
+					return nil, err
+				}
+				if err := d.Set("password_policy_name", passwordPolicyName.Name()); err != nil {
+					return nil, err
+				}
+				if err := d.Set("user_name", userName.Name()); err != nil {
+					return nil, err
+				}
+				return []*schema.ResourceData{d}, nil
+			},
+		},
+	}
+}
+
+func CreateUserPasswordPolicyAttachment(d *schema.ResourceData, meta interface{}) error {
+	db := meta.(*sql.DB)
+	client := sdk.NewClientFromDB(db)
+	ctx := context.Background()
+
+	userName := sdk.NewAccountObjectIdentifierFromFullyQualifiedName(d.Get("user_name").(string))
+	passwordPolicyDatabase := sdk.NewAccountIdentifierFromFullyQualifiedName(d.Get("password_policy_database").(string))
```

Comment:
if using three separate parts, you could just do sdk.NewSchemaObjectIdcentifier(db, schema, name)